varys/consumer.py

```diff
@@ -58,7 +58,8 @@ def _reconnect(self):
             self.stop()
             self._log.warning(f"Reconnecting after {self._reconnect_delay} seconds")
             time.sleep(self._reconnect_delay)
-            self._connect()
+            self._connection = self._connect()
+            self._connection.ioloop.start()
```

Comment:
The Pika [example asynchronous consumer](https://github.com/pika/pika/blob/main/examples/asynchronous_consumer_example.py) in effect reconnects by creating a new consumer. That will in effect execute [`run`](https://github.com/pika/pika/blob/f4d8f8ff02a4da4653749c86161b7d52e53f73fe/examples/asynchronous_consumer_example.py#L364-L370), so that's the code I've duplicated here.